faiss/python/__init__.py

```diff
@@ -96,6 +96,21 @@ def replacement_decode(self, codes):
 handle_Quantizer(ScalarQuantizer)
 
 
+def handle_NSG(the_class):
+
+    def replacement_build(self, x, graph):
+        n, d = x.shape
+        graph = graph.reshape((-1,))
+        graph = np.ascontiguousarray(graph)
```

Comment:
It is the caller's responsibility to provide a contiguous array (this is checked in swig_ptr)